static/app/components/events/interfaces/frame/context.tsx

```diff
@@ -40,6 +50,51 @@ type Props = {
   registersMeta?: Record<any, any>;
 };
 
+export function getCoverageColors(
+  organization: Organization,
+  missingData: boolean,
+  lines: [number, string][],
+  lineCoverage?: LineCoverage[]
+): string[] {
+  const shouldShowCodecovData =
+    organization?.features.includes('codecov-stacktrace-integration') &&
+    organization?.codecovAccess;
+
+  if (!shouldShowCodecovData || !missingData || !lineCoverage) {
+    return lines.map(() => 'transparent');
+  }
+
+  let coverageIndex = 0;
+  for (const [index, lc] of lineCoverage.entries()) {
+    if (lc.lineNo === lines[0][0]) {
+      coverageIndex = index;
+      break;
+    }
+  }
+
+  return lines.map(line => {
+    const coverageLine = lineCoverage[coverageIndex];
```

Comment:
you might consider converting lineCoverage to an object and accessing the line you want by key that way?
we have a lodash/keyBy that would handle this. You're already doing a .entries on lineCoverage which is just as wasteful
```
import keyBy from 'lodash/keyBy';
```